.github/workflows/builddockerimage.yml

```diff
@@ -0,0 +1,79 @@
+name: builddockerimage
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        required: true
+        type: string
+      goversion:
+        required: true
+        type: string
+      goargs:
+        required: true
+        type: string
+      labels:
+        required: true
+        type: string
+      version:
+        required: true
+        type: string
+
+jobs:
+  build_job:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: amd64
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Set up Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ inputs.goversion }}
+      - name: Setup cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ./third_party_licenses
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-build-
+      - name: Prepare
+        run: |
+          # Get licenses if no cache exists
```

Comment:
should probably be changed to separate file after #783 is merged

Reply:
ok